.azuredevops/pipelineTemplates/module.jobs.deploy.yml

```diff
@@ -0,0 +1,287 @@
+              azurePowerShellVersion: ${{ parameters.azurePowerShellVersion }}
+              preferredAzurePowerShellVersion: ${{ parameters.preferredAzurePowerShellVersion }}
+              ScriptType: InlineScript
+              pwsh: true
+              inline: |
+                # Load used functions
+                . (Join-Path '$(moduleRepoRoot)' 'utilities' 'pipelines' 'tokensReplacement' 'Convert-TokensInParameterFile.ps1')
+
+                # Load Settings File
+                $Settings = Get-Content -Path (Join-Path '$(moduleRepoRoot)' 'settings.json') | ConvertFrom-Json
+
+                # Initialize Default Parameter File Tokens
+                $DefaultParameterFileTokens = @(
+                    @{ Name = 'resourceGroupName'; Value = '${{ parameters.resourceGroupName }}' }
+                    @{ Name = 'subscriptionId';    Value = '${{ parameters.subscriptionId }}' }
+                    @{ Name = 'managementGroupId'; Value = '${{ parameters.managementGroupId }}' }
+                    @{ Name = "tenantId";          Value = '$(ARM_TENANT_ID)' }
+                    @{ Name = "deploymentSpId";    Value = '$(DEPLOYMENT_SP_ID)' }
+                ) | ForEach-Object { [PSCustomObject]$PSItem }
+
+                # Construct Token Function Input
+                $ConvertTokensInputs = @{
+                    ParameterFilePath                 = Join-Path '$(parametersRepoRoot)' '${{ deploymentBlock.path }}'
+                    DefaultParameterFileTokens        = $DefaultParameterFileTokens
+                    LocalCustomParameterFileTokens    = $Settings.parameterFileTokens.localTokens.tokens
+                    TokenPrefix                       = $Settings.parameterFileTokens.tokenPrefix
+                    TokenSuffix                       = $Settings.parameterFileTokens.tokenSuffix
+                }
+                # Invoke Token Replacement Functionality
+                $null = Convert-TokensInParameterFile @ConvertTokensInputs -Verbose
+
+          # [Validation] task(s)
+          #---------------------
+          - task: AzurePowerShell@5
+            displayName: 'Validate [${{ deploymentBlock.path }}] via connection [${{ parameters.serviceConnection }}]'
+            inputs:
+              azureSubscription: ${{ parameters.serviceConnection }}
+              azurePowerShellVersion: ${{ parameters.azurePowerShellVersion }}
+              preferredAzurePowerShellVersion: ${{ parameters.preferredAzurePowerShellVersion }}
+              ScriptType: InlineScript
+              pwsh: true
+              inline: |
+                # Load used functions
+                . (Join-Path '$(moduleRepoRoot)' '$(pipelineFunctionsPath)' 'resourceValidation' 'Test-TemplateWithParameterFile.ps1')
+
+                $templateFilePath = ([String]::IsNullOrEmpty('${{ deploymentBlock.templateFilePath }}')) ?
+                  (Join-Path '$(ModuleRepoRoot)' '$(modulePath)' 'deploy.bicep') :
```

Comment:
Can we make this more interoperable with json too?

Reply:
done